### ISSUE TITLE
`bad-string-format-type` (E1307) false positives for `%i`, `%u` and `%a`

### ISSUE THREAD

**Issue by Synvoya** [PyCQA/pylint#11147]
### Bug description

`bad-string-format-type` (`E1307`) reports false positives for valid `%`-formatting: `%i` / `%u` applied to a float, and `%a` applied to any non-int type. All four cases below succeed at runtime.

```python
FLOAT_VAL = 3.5
"%i" % FLOAT_VAL   # -> '3'        (falsely flagged)
"%u" % FLOAT_VAL   # -> '3'        (falsely flagged)
"%a" % FLOAT_VAL   # -> '3.5'      (falsely flagged)
"%a" % "hello"     # -> "'hello'"  (falsely flagged)
```

### Command used

```shell
pylint --disable=all --enable=bad-string-format-type f.py
```

### Pylint output

```
E1307: Argument 'builtins.float' does not match format type 'i'
E1307: Argument 'builtins.float' does not match format type 'u'
E1307: Argument 'builtins.float' does not match format type 'a'
E1307: Argument 'builtins.str' does not match format type 'a'
```

### Expected behavior

No message — all four are valid. (The genuine error `"%x" % 1.1` is still correctly flagged.)

### Root cause

`pylint/checkers/strings.py` — `arg_matches_format_type`: the "always printable" early return covers only `s`/`r` (omits `a`, which is equally type-agnostic), and the float conversion table `"deEfFgGn%"` omits `i` and `u` (both truncate a float exactly like `d`).

### Pylint version

```
pylint main (2bfab5f), astroid 4.2.0b3, Python 3.12
```
